sbt 1.0.2 publish with credentials fails to push jar and sources artefacts (for one of three subprojects)

This one is weird, and I likely won't be able to formulate it according to the contribution guidelines. But I think it's worth reporting in order to see whether others are seeing similar cases on-field as well.

In short:

- upgrading one of my sbt projects (multiproject, with three subprojects) from sbt 0.13.15 to 1.0.2 causes one of the projects (always the same) to be only partially published (pom's are there, but jars and sources aren't) to the company's Nexus Maven repo

Screenshots: 

![screen shot 2017-09-21 at 11 41 58](https://user-images.githubusercontent.com/1297734/30741225-465f3634-9f94-11e7-8d0d-d44cf2aca391.png)
**Properly published subproject**

![screen shot 2017-09-21 at 11 41 36](https://user-images.githubusercontent.com/1297734/30741230-496d8df8-9f94-11e7-865d-cd875e5f61d9.png)
**Partial publish (no jars, no sources)**

This happens for both release and snapshot publishes.

All subprojects share the same credentials setup:

```
def publishAs(name: String) = {
  val nexus = "https://maven.myCompany.net/"

  Def.settings(
    moduleName := name,
    publishMavenStyle := true,
    publishArtifact in (Compile, packageDoc) := false,
    publishTo := {
      if (isSnapshot.value) {
        Some("snapshots" at nexus + "content/repositories/snapshots")
      } else {
        Some("releases" at nexus + "content/repositories/releases")
      }
    },
    credentials ++= (for {
      user <- Option(System.getenv().get("MAVEN_USER"))
      pw <- Option(System.getenv().get("MAVEN_PASSWORD"))
    } yield Credentials("Sonatype Nexus Repository Manager", "maven.myCompany.net", user, pw)).toSeq
  )
}
```

All of this works, with 0.13.15 in `project/build.properties`. Changing to 1.0.2 makes publishing consistently fail.

On the command line one sees:

`[error] (testkit/*:publish) java.io.IOException: PUT operation to URL https://maven.myCompany.net/content/repositories/snapshots/fi/myCompany/basement/cloak-akka-http-testkit_2.12/0.0.13-SNAPSHOT/cloak-akka-http-testkit_2.12-0.0.13-SNAPSHOT.jar failed with status code 502: Bad Gateway
`

The Nexus log show `401` (Unauthorized) for the failing parts.

My current work-around is simply to not upgrade to 1.0.2 within the particular project.

I'm seeing the same issue with downloading. It's finding the artifact, but it's failing on a 404 from a different repository

```
[debug] sbt-chain: Checking cache for: dependency: net.cakesolutions#scala-kafka-client_2.11;0.11.0.0 {compile=[default(compile)]}
[debug] sbt-chain-delegate: Checking cache for: dependency: net.cakesolutions#scala-kafka-client_2.11;0.11.0.0 {compile=[default(compile)]}
[debug]                 tried /home/jenkins/.ivy2/local/net.cakesolutions/scala-kafka-client_2.11//0.11.0.0/ivys/ivy.xml
[debug]         local: no ivy file found for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 tried https://repo1.maven.org/maven2/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug] CLIENT ERROR: Not Found url=https://repo1.maven.org/maven2/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         public: no ivy file found for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 tried https://oss.sonatype.org/content/repositories/public/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug] CLIENT ERROR: Not Found url=https://oss.sonatype.org/content/repositories/public/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         sonatype-public: no ivy file found for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 tried https://<company-artifactory>/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug] CLIENT ERROR: Not Found url=https://<company-artifactory>/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         <company-artifactory> releases: no ivy file found for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 tried https://<company-artifactory>/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug] CLIENT ERROR: Conflict url=https://<company-artifactory>/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         <company-artifactory> snapshots: no ivy file found for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 tried https://<company-artifactory>/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug] CLIENT ERROR: Not Found url=https://<company-artifactory>/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         <company-artifactory> Third Party: no ivy file found for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 tried https://dl.bintray.com/cakesolutions/maven/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         bintray-cakesolutions-maven: found md file for net.cakesolutions#scala-kafka-client_2.11;0.11.0.0
[debug]                 => https://dl.bintray.com/cakesolutions/maven/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom (0.11.0.0)
[debug] downloading https://dl.bintray.com/cakesolutions/maven/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom ...
[debug]         bintray-cakesolutions-maven: downloading https://dl.bintray.com/cakesolutions/maven/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         bintray-cakesolutions-maven: downloading https://dl.bintray.com/cakesolutions/maven/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom.sha1
[debug] sha1 OK for https://dl.bintray.com/cakesolutions/maven/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.pom
[debug]         [SUCCESSFUL ] net.cakesolutions#scala-kafka-client_2.11;0.11.0.0!scala-kafka-client_2.11.pom(pom.original) (297ms)
[debug]                 tried /home/jenkins/.ivy2/local/net.cakesolutions/scala-kafka-client_2.11//0.11.0.0/jars/scala-kafka-client_2.11.jar
[debug]                 tried https://repo1.maven.org/maven2/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.jar
[debug] CLIENT ERROR: Not Found url=https://repo1.maven.org/maven2/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.jar
[debug]                 tried https://oss.sonatype.org/content/repositories/public/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.jar
[debug] problem occurred while resolving dependency: net.cakesolutions#scala-kafka-client_2.11;0.11.0.0 {compile=[default(compile)]} with bintray-cakesolutions-maven: java.lang.RuntimeException: sonatype-public: unable to get resource for net/cakesolutions#scala-kafka-client_2.11;0.11.0.0: res=https://oss.sonatype.org/content/repositories/public/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.jar: java.net.ProtocolException: Unexpected status line: 0
```

Full stack trace

```
debug] problem occurred while resolving dependency: net.cakesolutions#scala-kafka-client_2.11;0.11.0.0 {compile=[default(compile)]} with bintray-cakesolutions-maven: java.lang.RuntimeException: sonatype-public: unable to get resource for net/cakesolutions#scala-kafka-client_2.11;0.11.0.0: res=https://oss.sonatype.org/content/repositories/public/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.jar: java.net.ProtocolException: Unexpected status line: 0
[debug]         at org.apache.ivy.plugins.resolver.RepositoryResolver.findResourceUsingPattern(RepositoryResolver.java:135)
[debug]         at org.apache.ivy.plugins.resolver.AbstractPatternsBasedResolver.findResourceUsingPatterns(AbstractPatternsBasedResolver.java:96)
[debug]         at org.apache.ivy.plugins.resolver.IBiblioResolver.findArtifactRef(IBiblioResolver.java:124)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.getArtifactRef(BasicResolver.java:1031)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.locate(BasicResolver.java:893)
[debug]         at org.apache.ivy.plugins.resolver.IBiblioResolver.locate(IBiblioResolver.java:553)
[debug]         at org.apache.ivy.plugins.resolver.ChainResolver.locate(ChainResolver.java:298)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.locate(SbtChainResolver.scala:82)
[debug]         at org.apache.ivy.plugins.resolver.ChainResolver.locate(ChainResolver.java:298)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.locate(SbtChainResolver.scala:82)
[debug]         at org.apache.ivy.plugins.parser.m2.PomModuleDescriptorParser.addSourcesAndJavadocArtifactsIfPresent(PomModuleDescriptorParser.java:331)
[debug]         at org.apache.ivy.plugins.parser.m2.PomModuleDescriptorParser.parseDescriptor(PomModuleDescriptorParser.java:307)
[debug]         at org.apache.ivy.plugins.parser.m2.PomModuleDescriptorParser.parseDescriptor(PomModuleDescriptorParser.java:112)
[debug]         at sbt.internal.librarymanagement.CustomPomParser.parseDescriptor(CustomPomParser.scala:43)
[debug]         at org.apache.ivy.core.cache.DefaultRepositoryCacheManager$MyModuleDescriptorProvider.provideModule(DefaultRepositoryCacheManager.java:716)
[debug]         at org.apache.ivy.core.cache.ModuleDescriptorMemoryCache.getStale(ModuleDescriptorMemoryCache.java:68)
[debug]         at org.apache.ivy.core.cache.DefaultRepositoryCacheManager.getStaledMd(DefaultRepositoryCacheManager.java:733)
[debug]         at org.apache.ivy.core.cache.DefaultRepositoryCacheManager.cacheModuleDescriptor(DefaultRepositoryCacheManager.java:1195)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.parse(BasicResolver.java:536)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.getDependency(BasicResolver.java:271)
[debug]         at org.apache.ivy.plugins.resolver.IBiblioResolver.getDependency(IBiblioResolver.java:538)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$$anonfun$defaultConvert$lzycompute$1$PluginCapableResolver$1.sbt$internal$librarymanagement$ConvertResolver$DescriptorRequired$$super$getDependency(ConvertResolver.scala:164)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$DescriptorRequired.getDependency(ConvertResolver.scala:291)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$DescriptorRequired.getDependency$(ConvertResolver.scala:288)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$$anonfun$defaultConvert$lzycompute$1$PluginCapableResolver$1.getDependency(ConvertResolver.scala:164)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.performResolution$1(SbtChainResolver.scala:152)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getResults$3(SbtChainResolver.scala:172)
[debug]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[debug]         at scala.collection.Iterator.foreach(Iterator.scala:929)
[debug]         at scala.collection.Iterator.foreach$(Iterator.scala:929)
[debug]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[debug]         at scala.collection.IterableLike.foreach(IterableLike.scala:71)
[debug]         at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
[debug]         at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[debug]         at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[debug]         at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[debug]         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getResults(SbtChainResolver.scala:165)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$lzycompute$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getDependency$1(SbtChainResolver.scala:323)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$resolveByAllMeans$2(SbtChainResolver.scala:249)
[debug]         at scala.Option.orElse(Option.scala:289)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.resolveByAllMeans(SbtChainResolver.scala:247)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getDependency(SbtChainResolver.scala:324)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.getDependency(SbtChainResolver.scala:90)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.performResolution$1(SbtChainResolver.scala:152)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getResults$3(SbtChainResolver.scala:172)
[debug]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[debug]         at scala.collection.Iterator.foreach(Iterator.scala:929)
[debug]         at scala.collection.Iterator.foreach$(Iterator.scala:929)
[debug]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[debug]         at scala.collection.IterableLike.foreach(IterableLike.scala:71)
[debug]         at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
[debug]         at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[debug]         at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[debug]         at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[debug]         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getResults(SbtChainResolver.scala:165)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$lzycompute$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getDependency$1(SbtChainResolver.scala:323)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$resolveByAllMeans$2(SbtChainResolver.scala:249)
[debug]         at scala.Option.orElse(Option.scala:289)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.resolveByAllMeans(SbtChainResolver.scala:247)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getDependency(SbtChainResolver.scala:324)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.getDependency(SbtChainResolver.scala:90)
[debug]         at org.apache.ivy.core.resolve.IvyNode.loadData(IvyNode.java:169)
[debug]         at org.apache.ivy.core.resolve.VisitNode.loadData(VisitNode.java:292)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.fetchDependencies(ResolveEngine.java:714)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.doFetchDependencies(ResolveEngine.java:799)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.fetchDependencies(ResolveEngine.java:722)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.getDependencies(ResolveEngine.java:594)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.resolve(ResolveEngine.java:234)
[debug]         at org.apache.ivy.Ivy.resolve(Ivy.java:517)
[debug]         at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:319)
[debug]         at sbt.internal.librarymanagement.IvyActions$.$anonfun$updateEither$1(IvyActions.scala:205)
[debug]         at sbt.internal.librarymanagement.IvySbt$Module.$anonfun$withModule$1(Ivy.scala:229)
[debug]         at sbt.internal.librarymanagement.IvySbt.$anonfun$withIvy$1(Ivy.scala:190)
[debug]         at sbt.internal.librarymanagement.IvySbt.sbt$internal$librarymanagement$IvySbt$$action$1(Ivy.scala:70)
[debug]         at sbt.internal.librarymanagement.IvySbt$$anon$3.call(Ivy.scala:77)
[debug]         at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
[debug]         at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:78)
[debug]         at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:97)
[debug]         at xsbt.boot.Using$.withResource(Using.scala:10)
[debug]         at xsbt.boot.Using$.apply(Using.scala:9)
[debug]         at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
[debug]         at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
[debug]         at xsbt.boot.Locks$.apply0(Locks.scala:31)
[debug]         at xsbt.boot.Locks$.apply(Locks.scala:28)
[debug]         at sbt.internal.librarymanagement.IvySbt.withDefaultLogger(Ivy.scala:77)
[debug]         at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:185)
[debug]         at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:182)
[debug]         at sbt.internal.librarymanagement.IvySbt$Module.withModule(Ivy.scala:228)
[debug]         at sbt.internal.librarymanagement.IvyActions$.updateEither(IvyActions.scala:190)
[debug]         at sbt.librarymanagement.ivy.IvyDependencyResolution.update(IvyDependencyResolution.scala:20)
[debug]         at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[debug]         at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:38)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:91)
[debug]         at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:68)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$19(LibraryManagement.scala:104)
[debug]         at scala.util.control.Exception$Catch.apply(Exception.scala:224)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:104)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:87)
[debug]         at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
[debug]         at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:118)
[debug]         at sbt.Classpaths$.$anonfun$updateTask$5(Defaults.scala:2353)
[debug]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[debug]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:42)
[debug]         at sbt.std.Transform$$anon$4.work(System.scala:64)
[debug]         at sbt.Execute.$anonfun$submit$2(Execute.scala:257)
[debug]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[debug]         at sbt.Execute.work(Execute.scala:266)
[debug]         at sbt.Execute.$anonfun$submit$1(Execute.scala:257)
[debug]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:167)
[debug]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:32)
[debug]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[debug]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[debug]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[debug]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[debug]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[debug]         at java.lang.Thread.run(Thread.java:748)
[debug] Caused by: java.net.ProtocolException: Unexpected status line: 0
[debug]         at okhttp3.internal.http.StatusLine.parse(StatusLine.java:69)
[debug]         at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
[debug]         at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:75)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
[debug]         at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
[debug]         at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
[debug]         at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
[debug]         at okhttp3.RealCall.execute(RealCall.java:69)
[debug]         at sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.getURLInfo(GigahorseUrlHandler.scala:48)
[debug]         at sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.getURLInfo(GigahorseUrlHandler.scala:29)
[debug]         at org.apache.ivy.util.url.URLHandlerDispatcher.getURLInfo(URLHandlerDispatcher.java:66)
[debug]         at org.apache.ivy.plugins.repository.url.URLResource.init(URLResource.java:65)
[debug]         at org.apache.ivy.plugins.repository.url.URLResource.exists(URLResource.java:81)
[debug]         at org.apache.ivy.plugins.resolver.RepositoryResolver.findResourceUsingPattern(RepositoryResolver.java:97)
[debug]         ... 121 more
[debug] problem occurred while resolving dependency: net.cakesolutions#scala-kafka-client_2.11;0.11.0.0 {compile=[default(compile)]} with sbt-chain-delegate: java.lang.RuntimeException: sonatype-public: unable to get resource for net/cakesolutions#scala-kafka-client_2.11;0.11.0.0: res=https://oss.sonatype.org/content/repositories/public/net/cakesolutions/scala-kafka-client_2.11/0.11.0.0/scala-kafka-client_2.11-0.11.0.0.jar: java.net.ProtocolException: Unexpected status line: 0
[debug]         at org.apache.ivy.plugins.resolver.RepositoryResolver.findResourceUsingPattern(RepositoryResolver.java:135)
[debug]         at org.apache.ivy.plugins.resolver.AbstractPatternsBasedResolver.findResourceUsingPatterns(AbstractPatternsBasedResolver.java:96)
[debug]         at org.apache.ivy.plugins.resolver.IBiblioResolver.findArtifactRef(IBiblioResolver.java:124)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.getArtifactRef(BasicResolver.java:1031)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.locate(BasicResolver.java:893)
[debug]         at org.apache.ivy.plugins.resolver.IBiblioResolver.locate(IBiblioResolver.java:553)
[debug]         at org.apache.ivy.plugins.resolver.ChainResolver.locate(ChainResolver.java:298)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.locate(SbtChainResolver.scala:82)
[debug]         at org.apache.ivy.plugins.resolver.ChainResolver.locate(ChainResolver.java:298)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.locate(SbtChainResolver.scala:82)
[debug]         at org.apache.ivy.plugins.parser.m2.PomModuleDescriptorParser.addSourcesAndJavadocArtifactsIfPresent(PomModuleDescriptorParser.java:331)
[debug]         at org.apache.ivy.plugins.parser.m2.PomModuleDescriptorParser.parseDescriptor(PomModuleDescriptorParser.java:307)
[debug]         at org.apache.ivy.plugins.parser.m2.PomModuleDescriptorParser.parseDescriptor(PomModuleDescriptorParser.java:112)
[debug]         at sbt.internal.librarymanagement.CustomPomParser.parseDescriptor(CustomPomParser.scala:43)
[debug]         at org.apache.ivy.core.cache.DefaultRepositoryCacheManager$MyModuleDescriptorProvider.provideModule(DefaultRepositoryCacheManager.java:716)
[debug]         at org.apache.ivy.core.cache.ModuleDescriptorMemoryCache.getStale(ModuleDescriptorMemoryCache.java:68)
[debug]         at org.apache.ivy.core.cache.DefaultRepositoryCacheManager.getStaledMd(DefaultRepositoryCacheManager.java:733)
[debug]         at org.apache.ivy.core.cache.DefaultRepositoryCacheManager.cacheModuleDescriptor(DefaultRepositoryCacheManager.java:1195)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.parse(BasicResolver.java:536)
[debug]         at org.apache.ivy.plugins.resolver.BasicResolver.getDependency(BasicResolver.java:271)
[debug]         at org.apache.ivy.plugins.resolver.IBiblioResolver.getDependency(IBiblioResolver.java:538)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$$anonfun$defaultConvert$lzycompute$1$PluginCapableResolver$1.sbt$internal$librarymanagement$ConvertResolver$DescriptorRequired$$super$getDependency(ConvertResolver.scala:164)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$DescriptorRequired.getDependency(ConvertResolver.scala:291)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$DescriptorRequired.getDependency$(ConvertResolver.scala:288)
[debug]         at sbt.internal.librarymanagement.ConvertResolver$$anonfun$defaultConvert$lzycompute$1$PluginCapableResolver$1.getDependency(ConvertResolver.scala:164)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.performResolution$1(SbtChainResolver.scala:152)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getResults$3(SbtChainResolver.scala:172)
[debug]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[debug]         at scala.collection.Iterator.foreach(Iterator.scala:929)
[debug]         at scala.collection.Iterator.foreach$(Iterator.scala:929)
[debug]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[debug]         at scala.collection.IterableLike.foreach(IterableLike.scala:71)
[debug]         at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
[debug]         at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[debug]         at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[debug]         at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[debug]         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getResults(SbtChainResolver.scala:165)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$lzycompute$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getDependency$1(SbtChainResolver.scala:323)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$resolveByAllMeans$2(SbtChainResolver.scala:249)
[debug]         at scala.Option.orElse(Option.scala:289)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.resolveByAllMeans(SbtChainResolver.scala:247)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getDependency(SbtChainResolver.scala:324)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.getDependency(SbtChainResolver.scala:90)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.performResolution$1(SbtChainResolver.scala:152)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getResults$3(SbtChainResolver.scala:172)
[debug]         at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[debug]         at scala.collection.Iterator.foreach(Iterator.scala:929)
[debug]         at scala.collection.Iterator.foreach$(Iterator.scala:929)
[debug]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
[debug]         at scala.collection.IterableLike.foreach(IterableLike.scala:71)
[debug]         at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
[debug]         at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[debug]         at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[debug]         at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[debug]         at scala.collection.AbstractTraversable.map(Traversable.scala:104)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getResults(SbtChainResolver.scala:165)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$lzycompute$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.results$1(SbtChainResolver.scala:321)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$getDependency$1(SbtChainResolver.scala:323)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.$anonfun$resolveByAllMeans$2(SbtChainResolver.scala:249)
[debug]         at scala.Option.orElse(Option.scala:289)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.resolveByAllMeans(SbtChainResolver.scala:247)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver$CustomSbtResolution$.getDependency(SbtChainResolver.scala:324)
[debug]         at sbt.internal.librarymanagement.ivyint.SbtChainResolver.getDependency(SbtChainResolver.scala:90)
[debug]         at org.apache.ivy.core.resolve.IvyNode.loadData(IvyNode.java:169)
[debug]         at org.apache.ivy.core.resolve.VisitNode.loadData(VisitNode.java:292)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.fetchDependencies(ResolveEngine.java:714)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.doFetchDependencies(ResolveEngine.java:799)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.fetchDependencies(ResolveEngine.java:722)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.getDependencies(ResolveEngine.java:594)
[debug]         at org.apache.ivy.core.resolve.ResolveEngine.resolve(ResolveEngine.java:234)
[debug]         at org.apache.ivy.Ivy.resolve(Ivy.java:517)
[debug]         at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:319)
[debug]         at sbt.internal.librarymanagement.IvyActions$.$anonfun$updateEither$1(IvyActions.scala:205)
[debug]         at sbt.internal.librarymanagement.IvySbt$Module.$anonfun$withModule$1(Ivy.scala:229)
[debug]         at sbt.internal.librarymanagement.IvySbt.$anonfun$withIvy$1(Ivy.scala:190)
[debug]         at sbt.internal.librarymanagement.IvySbt.sbt$internal$librarymanagement$IvySbt$$action$1(Ivy.scala:70)
[debug]         at sbt.internal.librarymanagement.IvySbt$$anon$3.call(Ivy.scala:77)
[debug]         at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
[debug]         at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:78)
[debug]         at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:97)
[debug]         at xsbt.boot.Using$.withResource(Using.scala:10)
[debug]         at xsbt.boot.Using$.apply(Using.scala:9)
[debug]         at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
[debug]         at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
[debug]         at xsbt.boot.Locks$.apply0(Locks.scala:31)
[debug]         at xsbt.boot.Locks$.apply(Locks.scala:28)
[debug]         at sbt.internal.librarymanagement.IvySbt.withDefaultLogger(Ivy.scala:77)
[debug]         at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:185)
[debug]         at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:182)
[debug]         at sbt.internal.librarymanagement.IvySbt$Module.withModule(Ivy.scala:228)
[debug]         at sbt.internal.librarymanagement.IvyActions$.updateEither(IvyActions.scala:190)
[debug]         at sbt.librarymanagement.ivy.IvyDependencyResolution.update(IvyDependencyResolution.scala:20)
[debug]         at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[debug]         at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:38)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:91)
[debug]         at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:68)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$19(LibraryManagement.scala:104)
[debug]         at scala.util.control.Exception$Catch.apply(Exception.scala:224)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:104)
[debug]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:87)
[debug]         at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
[debug]         at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:118)
[debug]         at sbt.Classpaths$.$anonfun$updateTask$5(Defaults.scala:2353)
[debug]         at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[debug]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:42)
[debug]         at sbt.std.Transform$$anon$4.work(System.scala:64)
[debug]         at sbt.Execute.$anonfun$submit$2(Execute.scala:257)
[debug]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[debug]         at sbt.Execute.work(Execute.scala:266)
[debug]         at sbt.Execute.$anonfun$submit$1(Execute.scala:257)
[debug]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:167)
[debug]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:32)
[debug]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[debug]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[debug]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[debug]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[debug]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[debug]         at java.lang.Thread.run(Thread.java:748)
[debug] Caused by: java.net.ProtocolException: Unexpected status line: 0
[debug]         at okhttp3.internal.http.StatusLine.parse(StatusLine.java:69)
[debug]         at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
[debug]         at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:75)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
[debug]         at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
[debug]         at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
[debug]         at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
[debug]         at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
[debug]         at okhttp3.RealCall.execute(RealCall.java:69)
[debug]         at sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.getURLInfo(GigahorseUrlHandler.scala:48)
[debug]         at sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.getURLInfo(GigahorseUrlHandler.scala:29)
[debug]         at org.apache.ivy.util.url.URLHandlerDispatcher.getURLInfo(URLHandlerDispatcher.java:66)
[debug]         at org.apache.ivy.plugins.repository.url.URLResource.init(URLResource.java:65)
[debug]         at org.apache.ivy.plugins.repository.url.URLResource.exists(URLResource.java:81)
[debug]         at org.apache.ivy.plugins.resolver.RepositoryResolver.findResourceUsingPattern(RepositoryResolver.java:97)
[debug]         ... 121 more
```

--- a/build.sbt
+++ b/build.sbt
@@ -42,7 +42,9 @@ def commonSettings: Seq[Setting[_]] = Def.settings(
       case _                           => old ++ List("-Ywarn-unused", "-Ywarn-unused-import", "-YdisableFlatCpCaching")
     }
   },
-  inCompileAndTest(scalacOptions in console --= Vector("-Ywarn-unused-import", "-Ywarn-unused", "-Xlint")),
+  inCompileAndTest(
+    scalacOptions in console --= Vector("-Ywarn-unused-import", "-Ywarn-unused", "-Xlint")
+  ),
   scalafmtOnCompile := true,
   Test / scalafmtOnCompile := true,
   publishArtifact in Compile := true,
@@ -52,13 +54,22 @@ def commonSettings: Seq[Setting[_]] = Def.settings(
 
 val mimaSettings = Def settings (
   mimaPreviousArtifacts := Set(
-    "1.0.0", "1.0.1", "1.0.2", "1.0.3", "1.0.4",
-    "1.1.0", "1.1.1", "1.1.2", "1.1.3", "1.1.4",
+    "1.0.0",
+    "1.0.1",
+    "1.0.2",
+    "1.0.3",
+    "1.0.4",
+    "1.1.0",
+    "1.1.1",
+    "1.1.2",
+    "1.1.3",
+    "1.1.4",
     "1.2.0",
-  ) map (version =>
-    organization.value %% moduleName.value % version
-      cross (if (crossPaths.value) CrossVersion.binary else CrossVersion.disabled)
-  ),
+  ) map (
+      version =>
+        organization.value %% moduleName.value % version
+          cross (if (crossPaths.value) CrossVersion.binary else CrossVersion.disabled)
+    ),
 )
 
 lazy val lmRoot = (project in file("."))
@@ -73,7 +84,8 @@ lazy val lmRoot = (project in file("."))
           Some(ScmInfo(url(s"https://github.com/$slug"), s"git@github.com:$slug.git"))
         },
         bintrayPackage := "librarymanagement",
-      )),
+      )
+    ),
     commonSettings,
     name := "LM Root",
     publish := {},
@@ -134,33 +146,32 @@ lazy val lmCore = (project in file("core"))
     mimaBinaryIssueFilters ++= Seq(
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.EvictionWarningOptions.this"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.EvictionWarningOptions.copy"),
-      exclude[IncompatibleResultTypeProblem]("sbt.librarymanagement.EvictionWarningOptions.copy$default$7"),
-
+      exclude[IncompatibleResultTypeProblem](
+        "sbt.librarymanagement.EvictionWarningOptions.copy$default$7"
+      ),
       // internal class moved
       exclude[MissingClassProblem]("sbt.internal.librarymanagement.InlineConfigurationFunctions"),
       // dropped internal class parent (InlineConfigurationFunctions)
       exclude[MissingTypesProblem]("sbt.librarymanagement.ModuleDescriptorConfiguration$"),
-
       // Configuration's copy method was never meant to be public
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.Configuration.copy"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.Configuration.copy$default$*"),
-
       // the data type copy methods were never meant to be public
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.copy"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.copy$default$*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleReportExtra.copy"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleReportExtra.copy$default$*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactTypeFilterExtra.copy"),
-      exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactTypeFilterExtra.copy$default$*"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.librarymanagement.ArtifactTypeFilterExtra.copy$default$*"
+      ),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleIDExtra.copy"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleIDExtra.copy$default$*"),
-
       // these abstract classes are private[librarymanagement] so it's fine if they have more methods
       exclude[ReversedMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.*"),
       exclude[ReversedMissingMethodProblem]("sbt.librarymanagement.ModuleReportExtra.*"),
       exclude[ReversedMissingMethodProblem]("sbt.librarymanagement.ArtifactTypeFilterExtra.*"),
       exclude[ReversedMissingMethodProblem]("sbt.librarymanagement.ModuleIDExtra.*"),
-
       // these abstract classes are private[librarymanagement] so they can lose these abstract methods
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.type"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.url"),
@@ -170,9 +181,10 @@ lazy val lmCore = (project in file("core"))
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.classifier"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactExtra.extension"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactTypeFilterExtra.types"),
-
       // contraband issue
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ConfigurationReportLite.copy*"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ConfigurationReportLite.copy*"
+      ),
       exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.UpdateReportLite.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.Artifact.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ArtifactTypeFilter.copy*"),
@@ -189,7 +201,9 @@ lazy val lmCore = (project in file("core"))
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.FileConfiguration.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.FileRepository.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.Full.copy*"),
-      exclude[DirectMissingMethodProblem]("sbt.librarymanagement.GetClassifiersConfiguration.copy*"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.librarymanagement.GetClassifiersConfiguration.copy*"
+      ),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.GetClassifiersModule.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.InclExclRule.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.IvyFileConfiguration.copy*"),
@@ -198,7 +212,9 @@ lazy val lmCore = (project in file("core"))
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.MavenCache.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.MavenRepo.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleConfiguration.copy*"),
-      exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleDescriptorConfiguration.copy*"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.librarymanagement.ModuleDescriptorConfiguration.copy*"
+      ),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleID.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleInfo.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ModuleReport.copy*"),
@@ -218,6 +234,10 @@ lazy val lmCore = (project in file("core"))
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.UpdateReport.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.UpdateStats.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.URLRepository.copy*"),
+      // private[sbt]
+      exclude[DirectMissingMethodProblem](
+        "sbt.librarymanagement.ResolverFunctions.useSecureResolvers"
+      ),
     ),
   )
   .configure(addSbtIO, addSbtUtilLogging, addSbtUtilPosition, addSbtUtilCache)
@@ -242,33 +262,68 @@ lazy val lmIvy = (project in file("ivy"))
       Vector("-Ywarn-unused-import", "-Ywarn-unused", "-Xlint"),
     mimaSettings,
     mimaBinaryIssueFilters ++= Seq(
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"),
-      exclude[IncompatibleMethTypeProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.checkStatusCode"),
-
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"
+      ),
+      exclude[IncompatibleMethTypeProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.checkStatusCode"
+      ),
       // sbt.internal methods that changed type signatures and aren't used elsewhere in production code
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.IvySbt#ParallelCachedResolutionResolveEngine.mergeErrors"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.IvySbt.cleanCachedResolutionCache"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.IvySbt#ParallelCachedResolutionResolveEngine.mergeErrors"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.IvySbt.cleanCachedResolutionCache"
+      ),
       exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.IvyRetrieve.artifacts"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.IvyScalaUtil.checkModule"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.CachedResolutionResolveEngine.mergeErrors"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.CachedResolutionResolveCache.buildArtificialModuleDescriptor"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.CachedResolutionResolveCache.buildArtificialModuleDescriptors"),
-      exclude[ReversedMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.CachedResolutionResolveEngine.mergeErrors"),
-
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"),
-      exclude[IncompatibleMethTypeProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.checkStatusCode"),
-
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.IvyScalaUtil.checkModule"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.CachedResolutionResolveEngine.mergeErrors"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.CachedResolutionResolveCache.buildArtificialModuleDescriptor"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.CachedResolutionResolveCache.buildArtificialModuleDescriptors"
+      ),
+      exclude[ReversedMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.CachedResolutionResolveEngine.mergeErrors"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"
+      ),
+      exclude[IncompatibleMethTypeProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler#SbtUrlInfo.this"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.checkStatusCode"
+      ),
       // contraband issue
-      exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ivy.ExternalIvyConfiguration.copy*"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.librarymanagement.ivy.ExternalIvyConfiguration.copy*"
+      ),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ivy.InlineIvyConfiguration.copy*"),
       exclude[DirectMissingMethodProblem]("sbt.librarymanagement.ivy.IvyPaths.copy*"),
-
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.urlFactory"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.http"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.open"),
-      exclude[DirectMissingMethodProblem]("sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.this"),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.urlFactory"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.http"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.open"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.ivyint.GigahorseUrlHandler.this"
+      ),
+      exclude[DirectMissingMethodProblem](
+        "sbt.internal.librarymanagement.CustomPomParser.versionRangeFlag"
+      ),
     ),
   )
 
@@ -278,11 +333,13 @@ lazy val lmScriptedTest = (project in file("scripted-test"))
     commonSettings,
     skip in publish := true,
     name := "scripted-test",
-    scriptedLaunchOpts := { scriptedLaunchOpts.value ++
-      Seq("-Xmx1024M", "-Dplugin.version=" + version.value)
+    scriptedLaunchOpts := {
+      scriptedLaunchOpts.value ++
+        Seq("-Xmx1024M", "-Dplugin.version=" + version.value)
     },
     scriptedBufferLog := false
-  ).enablePlugins(SbtScriptedIT)
+  )
+  .enablePlugins(SbtScriptedIT)
 
 // we are updating the nightly process, so we are commenting this out for now
 // addCommandAlias("scriptedIvy", Seq(
@@ -303,14 +360,16 @@ def customCommands: Seq[Setting[_]] = Seq(
   }
 )
 
-inThisBuild(Seq(
-  whitesourceProduct                   := "Lightbend Reactive Platform",
-  whitesourceAggregateProjectName      := "sbt-lm-master",
-  whitesourceAggregateProjectToken     := "9bde4ccbaab7401a91f8cda337af84365d379e13abaf473b85cb16e3f5c65cb6",
-  whitesourceIgnoredScopes             += "scalafmt",
-  whitesourceFailOnError               := sys.env.contains("WHITESOURCE_PASSWORD"), // fail if pwd is present
-  whitesourceForceCheckAllDependencies := true,
-))
+inThisBuild(
+  Seq(
+    whitesourceProduct := "Lightbend Reactive Platform",
+    whitesourceAggregateProjectName := "sbt-lm-master",
+    whitesourceAggregateProjectToken := "9bde4ccbaab7401a91f8cda337af84365d379e13abaf473b85cb16e3f5c65cb6",
+    whitesourceIgnoredScopes += "scalafmt",
+    whitesourceFailOnError := sys.env.contains("WHITESOURCE_PASSWORD"), // fail if pwd is present
+    whitesourceForceCheckAllDependencies := true,
+  )
+)
 
 def inCompileAndTest(ss: SettingsDefinition*): Seq[Setting[_]] =
   Seq(Compile, Test) flatMap (inConfig(_)(Def.settings(ss: _*)))

--- a/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala
+++ b/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala
@@ -10,6 +10,7 @@ package internal
 package librarymanagement
 
 import java.util.Locale
+import scala.util.control.NonFatal
 
 object LMSysProp {
   def booleanOpt(name: String): Option[Boolean] =
@@ -40,17 +41,21 @@ object LMSysProp {
 
   val useSecureResolvers: Boolean = getOrTrue("sbt.repository.secure")
 
-  def modifyVersionRange: Boolean = getOrTrue("sbt.modversionrange")
+  lazy val modifyVersionRange: Boolean = getOrTrue("sbt.modversionrange")
 
   lazy val isJavaVersion9Plus: Boolean = javaVersion > 8
   lazy val javaVersion: Int = {
-    // See Oracle section 1.5.3 at:
-    // https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html
-    val version = sys.props("java.specification.version").split("\\.").toList.map(_.toInt)
-    version match {
-      case 1 :: minor :: _ => minor
-      case major :: _      => major
-      case _               => 0
+    try {
+      // See Oracle section 1.5.3 at:
+      // https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html
+      val version = sys.props("java.specification.version").split("\\.").toList.map(_.toInt)
+      version match {
+        case 1 :: minor :: _ => minor
+        case major :: _      => major
+        case _               => 0
+      }
+    } catch {
+      case NonFatal(_) => 0
     }
   }
 

--- a/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala
+++ b/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala
@@ -1,0 +1,58 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+package internal
+package librarymanagement
+
+import java.util.Locale
+
+object LMSysProp {
+  def booleanOpt(name: String): Option[Boolean] =
+    sys.props.get(name).flatMap { x =>
+      x.toLowerCase(Locale.ENGLISH) match {
+        case "1" | "always" | "true" => Some(true)
+        case "0" | "never" | "false" => Some(false)
+        case "auto"                  => None
+        case _                       => None
+      }
+    }
+
+  def getOrFalse(name: String): Boolean = booleanOpt(name).getOrElse(false)
+  def getOrTrue(name: String): Boolean = booleanOpt(name).getOrElse(true)
+
+  // System property style:
+  //   1. use sbt. prefix
+  //   2. prefer short nouns
+  //   3. use dot for namespacing, and avoid making dot-separated English phrase
+  //   4. make active/enable properties, instead of "sbt.disable."
+  //
+  // Good: sbt.offline
+  //
+  // Bad:
+  // sbt.disable.interface.classloader.cache
+  // sbt.task.timings.on.shutdown
+  // sbt.skip.version.write -> sbt.genbuildprops=false
+
+  val useSecureResolvers: Boolean = getOrTrue("sbt.repository.secure")
+
+  def modifyVersionRange: Boolean = getOrTrue("sbt.modversionrange")
+
+  lazy val isJavaVersion9Plus: Boolean = javaVersion > 8
+  lazy val javaVersion: Int = {
+    // See Oracle section 1.5.3 at:
+    // https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html
+    val version = sys.props("java.specification.version").split("\\.").toList.map(_.toInt)
+    version match {
+      case 1 :: minor :: _ => minor
+      case major :: _      => major
+      case _               => 0
+    }
+  }
+
+  lazy val useGigahorse: Boolean = getOrTrue("sbt.gigahorse")
+}

--- a/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala
+++ b/core/src/main/scala/sbt/internal/librarymanagement/LMSysProp.scala
@@ -54,5 +54,5 @@ object LMSysProp {
     }
   }
 
-  lazy val useGigahorse: Boolean = getOrTrue("sbt.gigahorse")
+  lazy val useGigahorse: Boolean = getOrFalse("sbt.gigahorse")
 }

--- a/core/src/main/scala/sbt/internal/librarymanagement/SemanticSelectorExtra.scala
+++ b/core/src/main/scala/sbt/internal/librarymanagement/SemanticSelectorExtra.scala
@@ -4,6 +4,7 @@ import sbt.librarymanagement.VersionNumber
 import sbt.internal.librarymanagement.SemSelOperator.{ Lt, Lte, Gt, Gte, Eq }
 
 import scala.annotation.tailrec
+import java.util.Locale
 
 private[librarymanagement] abstract class SemSelAndChunkFunctions {
   protected def parse(andClauseToken: String): SemSelAndChunk = {
@@ -123,10 +124,11 @@ private[librarymanagement] abstract class SemComparatorExtra {
         // Identifiers consisting of only digits are compared numerically.
         // Numeric identifiers always have lower precedence than non-numeric identifiers.
         // Identifiers with letters are compared case insensitive lexical order.
-        case (true, true)   => implicitly[Ordering[Long]].compare(ts1head.toLong, ts2head.toLong)
-        case (false, true)  => 1
-        case (true, false)  => -1
-        case (false, false) => ts1head.toLowerCase.compareTo(ts2head.toLowerCase)
+        case (true, true)  => implicitly[Ordering[Long]].compare(ts1head.toLong, ts2head.toLong)
+        case (false, true) => 1
+        case (true, false) => -1
+        case (false, false) =>
+          ts1head.toLowerCase(Locale.ENGLISH).compareTo(ts2head.toLowerCase(Locale.ENGLISH))
       }
       if (cmp == 0) compareTags(ts1.tail, ts2.tail)
       else cmp

--- a/core/src/main/scala/sbt/librarymanagement/ResolverExtra.scala
+++ b/core/src/main/scala/sbt/librarymanagement/ResolverExtra.scala
@@ -95,8 +95,7 @@ private[sbt] class FakeRepository(resolver: AnyRef, name: String) extends xsbti.
 }
 
 private[librarymanagement] abstract class ResolverFunctions {
-  private[sbt] def useSecureResolvers =
-    sys.props.get("sbt.repository.secure") map { _.toLowerCase == "true" } getOrElse true
+  import sbt.internal.librarymanagement.LMSysProp.useSecureResolvers
 
   val TypesafeRepositoryRoot = typesafeRepositoryRoot(useSecureResolvers)
   val SbtRepositoryRoot = sbtRepositoryRoot(useSecureResolvers)

--- a/ivy/src/main/scala/sbt/internal/librarymanagement/CustomPomParser.scala
+++ b/ivy/src/main/scala/sbt/internal/librarymanagement/CustomPomParser.scala
@@ -242,9 +242,6 @@ object CustomPomParser {
         transform(dd, _ => newId)
       case None => dd
     }
-  private[sbt] lazy val versionRangeFlag = sys.props.get("sbt.modversionrange") map {
-    _.toLowerCase == "true"
-  } getOrElse true
 
   import collection.JavaConverters._
   def addExtra(
@@ -281,7 +278,7 @@ object CustomPomParser {
       addExtra(dd, dependencyExtra)
     }
     val withVersionRangeMod: Seq[DependencyDescriptor] =
-      if (versionRangeFlag) withExtra map { stripVersionRange } else withExtra
+      if (LMSysProp.modifyVersionRange) withExtra map { stripVersionRange } else withExtra
     val unique = IvySbt.mergeDuplicateDefinitions(withVersionRangeMod)
     unique foreach dmd.addDependency
 

--- a/ivy/src/main/scala/sbt/internal/librarymanagement/ivyint/ErrorMessageAuthenticator.scala
+++ b/ivy/src/main/scala/sbt/internal/librarymanagement/ivyint/ErrorMessageAuthenticator.scala
@@ -15,7 +15,8 @@ object ErrorMessageAuthenticator {
   private var securityWarningLogged = false
 
   private def originalAuthenticator: Option[Authenticator] = {
-    if (isJavaVersion9Plus) getDefaultAuthenticator else getTheAuthenticator
+    if (LMSysProp.isJavaVersion9Plus) getDefaultAuthenticator
+    else getTheAuthenticator
   }
 
   private[this] def getTheAuthenticator: Option[Authenticator] = {
@@ -100,15 +101,6 @@ object ErrorMessageAuthenticator {
       }
     doInstallIfIvy(originalAuthenticator)
   }
-
-  private[this] def isJavaVersion9Plus = javaVersion > 8
-  private[this] def javaVersion = {
-    // See Oracle section 1.5.3 at:
-    // https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html
-    val version = sys.props("java.specification.version").split("\\.").map(_.toInt)
-    if (version(0) == 1) version(1) else version(0)
-  }
-
 }
 
 /**

--- a/ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala
+++ b/ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala
@@ -4,6 +4,7 @@ package ivy
 import org.apache.ivy.plugins.resolver.DependencyResolver
 import org.apache.ivy.core.settings.IvySettings
 import sbt.util.Logger
+import sbt.internal.librarymanagement.LMSysProp
 
 /**
  * Represents configurable options for update task.
@@ -109,7 +110,7 @@ object UpdateOptions {
       interProjectFirst = true,
       latestSnapshots = true,
       cachedResolution = false,
-      gigahorse = sys.props.get("sbt.gigahorse") map { _.toLowerCase == "true" } getOrElse true,
+      gigahorse = LMSysProp.useGigahorse,
       resolverConverter = PartialFunction.empty,
       moduleResolvers = Map.empty
     )

